duke/src/main/java/DukeException.java

```diff
@@ -0,0 +1,7 @@
+public class DukeException extends Exception{
+    protected String type;
```

Comment:
why is this variable protected instead of private? You are not subclassing it. Same for other protected variables that don't need to be accessed by subclasses.